src/errors.rs

```diff
@@ -1,74 +1,247 @@
+use crate::{i18n::*,  lemmy_errors::LemmyErrorType};
+// use lemmy_api_common::error::*;
+use core::num::ParseIntError;
+use leptos::*;
 use serde::{Deserialize, Serialize};
 use serde_urlencoded::ser;
-use std::num::ParseIntError;
-use thiserror::Error;
+use strum_macros::Display;
 
-#[derive(Debug, Clone, Error, Serialize, Deserialize)]
-pub enum LemmyAppError {
-  #[error("Not Found")]
+pub type LemmyAppResult<T> = Result<T, LemmyAppError>;
+
+#[derive(Default, Display, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[serde(tag = "error", content = "message", rename_all = "snake_case")]
+pub enum LemmyAppErrorType {
+  #[default]
+  Unknown,
   NotFound,
-  #[error("Internal Server Error")]
   InternalServerError,
-  #[error("Couldn't parse params")]
+  InternalClientError,
   ParamsError,
-  #[error("{error:?}")]
-  APIError { error: String },
+
+  ApiError(LemmyErrorType),
+
+  EmptyUsername,
+  EmptyPassword,
+  MissingToken,
+
+  MissingReason,
 }
 
-impl LemmyAppError {
-  // pub fn status_code(&self) -> StatusCode {
-  //   match self {
-  //     LemmyAppError::NotFound => StatusCode::NOT_FOUND,
-  //     LemmyAppError::InternalServerError => StatusCode::INTERNAL_SERVER_ERROR,
-  //     Lemmy
-  //   }
-  // }
+pub fn message_from_error(error: &LemmyAppError) -> String {
+  let i18n = use_i18n();
+
+  let s = match error {
+    LemmyAppError {
+      error_type: LemmyAppErrorType::ApiError(LemmyErrorType::IncorrectLogin),
+      ..
+    } => t!(i18n, invalid_login)().to_string(),
+    LemmyAppError {
+      error_type: LemmyAppErrorType::EmptyUsername,
+      ..
+    } => t!(i18n, empty_username)().to_string(),
+    LemmyAppError {
+      error_type: LemmyAppErrorType::EmptyPassword,
+      ..
+    } => t!(i18n, empty_password)().to_string(),
+    LemmyAppError {
+      error_type: LemmyAppErrorType::MissingReason,
+      ..
+    } => t!(i18n, empty_reason)().to_string(),
+    LemmyAppError {
+      error_type: LemmyAppErrorType::InternalServerError,
+      ..
+    } => t!(i18n, internal)().to_string(),
+    LemmyAppError {
+      error_type: LemmyAppErrorType::Unknown,
+      ..
+    } => t!(i18n, unknown)().to_string(),
+    _ => "An error without description".to_string(),
+  };
```

Comment:
You'll likely find it easier if you match against `error.error_type` directly instead of destructuring it each time.

Reply:
i should have marked the PR as draft again before pushing update - this code may have been removed or changed